app/models/goal.py

```diff
@@ -2,4 +2,25 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    __tablename__ = 'goals'
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    tasks = db.relationship('Task', back_populates='goal', lazy=True)
+    
+    def to_dict(self):
+        return {
+                'id': self.id,
+                'title': self.title
+        }
+
+    @classmethod
+    def from_dict(cls, request_body):
+        goal = cls(
```

Comment:
Here, you're using `cls` to help make a new instance of `Goal`. In this case, I would personally argue that it might be beneficial to specify this as `Goal`... if we get into this function, we have reasonable certainty that `goal` should be an instance of `Goal`, no matter what `cls` is passed in.

I think there's a chance that Ada's curriculum teaches otherwise. ;) If that's the case, sorry for being contradictory, and I hope you think of this suggestion as just an interesting thought experiment.